pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository.
+	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
+
+	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
+	Jenkinsfile string `json:"jenkinsfile,omitempty"`
+
+	// Env contains additional environment variables you want to pass to the Jenkins Pipeline.
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
I thought our first iteration on an MVP here was just going to assume everything was in the Jenkinsfile, rather than trying to define a pipeline in the BuildConfig metadata.